src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,53 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+    private double randomNumber;
+
+    public Figure getRandomFigure() {
+        getRandomFigureNumber();
+        ColorSupplier colorSupplier = new ColorSupplier();
+        if (randomNumber == 0) {
+            return new Circle("RandomCircle",
+                    colorSupplier.getRandomColor().toString(),
+                    getRandomSide());
```

Comment:
move it to the private method
All private methods should be below public methods
```suggestion
            return prepareCircle();
```